mql.go

```diff
@@ -0,0 +1,103 @@
+package mql
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/iancoleman/strcase"
+)
+
+// WhereClause contains a SQL where clause condition and its arguments.
+type WhereClause struct {
+	// Condition is the where clause condition
+	Condition string
+	// Args for the where clause condition
+	Args []any
```

Comment:
Can we constrain this type?

Reply:
I don't think so, especially given that the caller can provide their own
validator/convertor.